# TASK 1: Permutations and Combinations



#### Importing nessesary library

In [69]:
# Mathematical functions from the standard library.
# https://docs.python.org/3/library/math.html
import math


#### Creating variables with amount of Cups

In [70]:
# Number of cups of tea in total.
nom_cups = 12

# Number of cups of tea with milk in first.
nom_cups_milk_first = 6

# Number of cups of tea with tea in first.
nom_cups_tea_first = 6

#### Use functions to calculate the number of ways to choose k elements from a set of n elements where the order of the chosen elements does not matter and no element is selected more than once

In [71]:
# Number of ways of selecting six cups from twelve.
total_com = math.comb(nom_cups, nom_cups_milk_first)

total_com

924

In [72]:
# Number of cups of tea in total.
n = 12

# Number of cups of tea with milk in first.
k = 6

In [73]:
# Probability of no errors (choosing all 6 milk-first cups)
prob_no_errors = 1 / total_com

In [74]:
# Probability of one error (choosing 5 milk-first cups and 1 tea-first cup)
choose_milk = math.comb(nom_cups_milk_first, 5)
choose_tea = math.comb(nom_cups_tea_first, 1)
prob_one_error = (choose_milk * choose_tea) / total_com

In [75]:
# Total probability of at most one error
prob_at_most_one_error = prob_no_errors + prob_one_error

In [76]:
# No of ways of selecting k objects from n without replacement and without order.
math.factorial(n) // (math.factorial(k) * math.factorial(n - k))

924

#### Conclusion: The probability of choosing first cup of milk is very low (1 of 924), so is unlikely lady will pick the correct cup. Accepting two errors would mean the person could still be right with 4 out of the 6 correct cups. The probability of achieving this would be calculated similarly by considering all combinations where 4 out of the 6 milk cups are chosen correctly and 2 out of the 6 tea cups are chosen incorrectly.